heapy-imp/src/test/scala/MySuite.scala

```diff
@@ -0,0 +1,141 @@
+class MySuite extends munit.FunSuite {
+    import AExp.*
+    import BExp.*
+    import ImpStmt.*
+    // x := 1;
+    // y := 0;
+    // while x <= 10 do {
+    //     y := y + x;
+    //     x := x + 1
+    // }
+    val i1 = Seq(
+      SeqExp(
+        // x := 1
+        Assign(AssignExp(Var("x"), Nat(1))),
+        Seq(
+          SeqExp(
+            // y := 0
+            Assign(AssignExp(Var("y"), Nat(0))),
+            While(
+              WhileExp(
+                // x <= 10
+                Comp(CompExp(Var("x"), Nat(10))),
+                Seq(
+                  SeqExp(
+                    // y := y + x
+                    Assign(
+                      AssignExp(Var("y"), Add(AddExp(Var("y"), Var("x"))))
+                    ),
+                    // x := x + 1
+                    Assign(AssignExp(Var("x"), Add(AddExp(Var("x"), Nat(1)))))
+                  )
+                )
+              )
+            )
+          )
+        )
+      )
+    )
+
+    // x := 1;
+    // counter := 0;
+    // while True do {
+    //     x := x + x;
+    //     counter := counter + 1;
+    //     if counter == 10 then break
+    // }
+    val i2 = build[ImpStmt](
+      "seq",
+      // x := 1
+      build[ImpStmt]("assign", "x".into, 1.into),
+      build[ImpStmt](
+        "seq",
+        // counter := 0
+        build[ImpStmt]("assign", "counter".into, 0.into),
+        // while True
+        build[ImpStmt](
+          "while",
+          BExp.True,
+          build[ImpStmt](
+            "seq",
+            // x := x + x
+            build[ImpStmt](
+              "assign",
+              "x".into,
+              build[AExp]("add", "x".into, "x".into)
+            ),
+            build[ImpStmt](
+              "seq",
+              // counter := counter + 1
+              build[ImpStmt](
+                "assign",
+                "counter".into,
+                build[AExp]("add", "counter".into, 1.into)
+              ),
+              // if 10 <= counter then break else skip
+              build[ImpStmt](
+                "cond",
+                build[BExp]("comp", 10.into, "counter".into),
+                ImpStmt.Break,
+                ImpStmt.Skip
+              )
+            )
+          )
+        )
+      )
+    )
+
+    // X := new(114514);
+    // Y = X
+    // *Y := 1919810
+    val i3 = build[ImpStmt](
```

Comment:
see the detailed use case for ref/deref here - you could expect the same ref/deref behavior just as any other language.